src/lib/tls/tls13_pqc/kex_to_kem_adapter.cpp

```diff
@@ -152,14 +112,21 @@ class KEX_to_KEM_Adapter_Encryption_Operation final : public PK_Ops::KEM_Encrypt
 
       size_t raw_kem_shared_key_length() const override { return kex_shared_key_length(m_public_key); }
 
-      size_t encapsulated_key_length() const override { return kex_public_value(m_public_key).size(); }
+      size_t encapsulated_key_length() const override {
+         // Serializing the public value into a short-lived heap-allocated
+         // vector is not ideal.
+         //
+         // TODO: Find a way to get the public value length without copying
+         //       the public value into a vector.
+         return m_public_key.raw_public_key_bits().size();
```

Comment:
Couldn’t you just call this once in the constructor and save the result?

Reply:
Certainly. Though, I don't think that'll help much. Most typical use cases will likely call this exactly once.
I somewhat had "point 5" of #3706 in mind when adding this TODO. I.e. providing a library-wide internal API to provide such magic numbers consistently. Though, admittedly, I didn't really look into the feasibility of it so far.

Reply:
I interpreted the comment more as indicating that the overhead of encoding/etc was wasteful in this context, thus my suggestion for caching the value.